### PR TITLE
Replace use_original_dst with original_dst listener filter

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Please provide a description for what this PR is for.

[use_original_dst  is deprecated. Use an original_dst listener filter instead](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/lds.proto)

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener_builder.go [hunk 1]
```diff
@@ -183,8 +174,6 @@ func (builder *ListenerBuilder) aggregateVirtualInboundListener() *ListenerBuild
 func NewListenerBuilder(node *model.Proxy) *ListenerBuilder {
 	builder := &ListenerBuilder{
 		node: node,
-		// The extra inbound listener has no side effect for iptables that doesn't redirect to 15006
-		useInboundFilterChain: true,
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/listener_builder.go]: This field is redundant as it is always true

**Comment by hzxuzhonghu** [comment]
@lambdai Do we have to do it first in istio proxy? I can see from the proxy logs that the outbound traffic is directly redirected to `PassthroughCluster` with this pr.

**Comment by howardjohn** [comment]
@lambdai can you review?

**Comment by hzxuzhonghu** [comment]
This won't work

**Pull request closed without merging**